### ISSUE TITLE
Support k8s Ingress resources

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#282]
If a user sets up its cluster in a similar way:
```
+--------+       +--------------------+         +-----------------+
|        |       |cilium --lb -d eth0 |         |  cilium -t vxlan|
|internet|       |         A          |         |          B      |
|        +------->eth0            eth1<--------->eth0             |
+--------+       +--------------------+         +-----------------+
```
The internet packets are received in `B` from `A`, as expected, but unfortunately `cilium -t vxlan` mode is not ready to read those packets received from `A`.

Until this feature is implemented, as a workaround we suggest to setup as follows
```
+--------+       +--------------------+         +-----------------+
|        |       |cilium --lb -d eth0 |         |  cilium -d eth0 |
|internet|       |         A          |         |          B      |
|        +------->eth0            eth1<--------->eth0             |
+--------+       +--------------------+         +-----------------+
```

Thus, cilium on machine `B` will be able to receive packets from `A` and send them to the proper container running the backend container in `B`.

**Comment by tgraf** [comment]
TODO:
 - [x] Add `ENCAP_IFINDEX` support to `bpf_lb.c`
 - [x] Define schema to store revnat ID in tunnel header

**Comment by aanm** [comment]
For now cilium only supports to snoop (`--device`) and LB (`--lb`) on a single network device.
In the future we'll allow the following modes:
- [ ] LB only (--lb eth0)
- [ ] container networking only (--device eth0)
- [ ] LB + container networking on different interfaces (--lb eth0 --device eth1)
- [X] LB + container networking on same interface (--lb eth0 --device eth0)

**Comment by jenningsloy318** [comment]
Hi If this feature is working ,I tried run cilium with lb, but got error. 
 #cilium-agent --lb  ens33 --kvstore etcd --etcd http://localhost:2379 
2017-05-15T18:17:53+08:00 INFO      _ _ _
2017-05-15T18:17:53+08:00 INFO  ___|_| |_|_ _ _____
2017-05-15T18:17:53+08:00 INFO |  _| | | | | |     |
2017-05-15T18:17:53+08:00 INFO |___|_|_|_|___|_|_|_|
2017-05-15T18:17:53+08:00 INFO Cilium 0.8.90 ce14613 Sat, 13 May 2017 13:54:11 +0200 go version go1.8.1 linux/amd64
2017-05-15T18:17:53+08:00 INFO clang and kernel versions: OK!
2017-05-15T18:17:53+08:00 INFO linking environment: OK!
2017-05-15T18:17:54+08:00 INFO BPF Verifier: OK!
2017-05-15T18:17:54+08:00 INFO Valid label prefix configuration:
2017-05-15T18:17:54+08:00 INFO  - :id.
2017-05-15T18:17:54+08:00 INFO  - :io.cilium.
2017-05-15T18:17:54+08:00 INFO  - k8s:io.kubernetes.pod.namespace
2017-05-15T18:17:54+08:00 INFO Using etcd as key-value store
2017-05-15T18:17:54+08:00 INFO Waiting for etcd client to be ready...
2017-05-15T18:17:54+08:00 INFO Etcd client ready
2017-05-15T18:17:54+08:00 ERRO Error while initializing daemon: Unable to run LB mode with tunnel mode
2017-05-15T18:17:54+08:00 CRIT Error while creating daemon: Unable to run LB mode with tunnel mode


And previous, I run this node with tunel mode, after all , I switch this node to lb mode. right now I only want to run it as lb mode,how can i do it? 


OS: ubuntu 16.04
kubernetes: v1.6.3
etcd: 3.0.14

**Comment by tgraf** [comment]
> 2017-05-15T18:17:54+08:00 ERRO Error while initializing daemon: Unable to run LB mode with tunnel mode
> 2017-05-15T18:17:54+08:00 CRIT Error while creating daemon: Unable to run LB mode with tunnel mode
> 
> And previous, I run this node with tunel mode, after all , I switch this node to lb mode. right now I only want to run it as lb mode,how can i do it?

This feature is currently not implemented yet which is why cilium will quit with the above error. We are planning to include this feature in the upcoming 0.9 release.

**Comment by jenningsloy318** [comment]
@tgraf Got it Thanks.

But about the roadmap, when this featrue(0.9 release) can be rolled out ?

**Comment by aalemayhu** [comment]
@jenningsloy318 the milestone page for 0.9 has a date https://github.com/cilium/cilium/milestone/2

**Comment by jenningsloy318** [comment]
Hope we can get it on May 19.

**Comment by tgraf** [comment]
@mchalla What's your ability to work on this?

**Comment by mchalla** [comment]
I just came back from India. I will finish it this week. Thanks.

**Comment by tgraf** [comment]
Being addressed i #1171

**Comment by aanm** [comment]
@tgraf I don't think this was completely addressed in #1171 right?
